### PR TITLE
Speed rewrite

### LINKED ISSUE #8: Feature Request: Keyboard mode for MCDU
### General
----

I am submitting a: (select one) 
* [ ] Bug report
* [x] Feature request

### Details
----
#### Current Behavior (if applicable)

MCDU entrys only possible by mouse

#### Requested Behavior  (feature requests)

Enter a Keyboard mode where all key strokes go to the MCDU so you can use your keyboard to manipulate the MCDU.  
F1 through F6 for left LSK 1-6  
F7 through F12 for right LSK 1-6

This would make MCDU entries much easier.

### LINKED ISSUE #75: Feature Request - Missing pfd features
@hayden2000 

Done in this phase:
* ILS warning flash
* Flashing QNH/STD when above/below transition alt/level
* Pulsing alt box, when about to reach selected alt
* V1, VR, V2 indications on PFD
* Fix ILS location
* ILS message stops flashing too early, change trigger property
* Flaps (F and S) speeds
* Max flap speeds
* Green dot speeds

### LINKED ISSUE #78: Feature Request - APPR page on MCDU
Add an APPR phase page to the MCDU under PERF

**Additional context**
Just necessary stuff for pfd such as VAPP, DEST QNH, F, S, O, VLS, etc

### LINKED ISSUE #96: Bug/Feature Request - Fix speed logic on PFD
Speed behaves oddly and causes bugs with new PFD features, needs logic rewrite to more closely match FCOM/A320 behavior.

So far I've found these, but I'm sure more will emerge when I get around to testing it thoroughly.
- V-speeds not displayed until t/o thrust set (should be visible on ground)
- In managed mode, speed doesn't seem to react to t/o sequence quick enough

### PR DESCRIPTION
**Pull request by mattmaring**
Rewrite the speed logic to more accurately model the actual logic!

### Description of Changes

- Added IRS INIT, FUELPRED, PROG-TO/CLB/CRZ/DES
- Improved INIT A, INIT B, PERF-TO/CLB/CRZ/DES/APPR/GA
- Add MCDU keyboard mode
- Implemented new phase progression, allows for manual approach/go-around + ability to revert with PROG page
- Implement new speed progression, central speed calcs in the mcdu
- Add speed protection bars (both normal and alt/direct law)
- Basic flightplan integration (will continue this work once this stage is merged)

### Bugs/Feature requests fixed
Fixes issues #8, #75, #78, and #96

### Checklist:
* [x] My changes are ready for merging.

### CODE REVIEW

**Review by merspieler — CHANGES_REQUESTED** [review]
Disregard, I've messed up.

File: A320-main.xml [hunk 1]
```diff
@@ -1443,6 +1444,39 @@
 					</script>
 				</binding>
 			</key>
+			<key n="8">
+                <name>Ctrl-H</name>
```
Comment (legoboyvdlp) [inline A320-main.xml:1448]: Careful with the tabs here on this one :)

**Review by legoboyvdlp — APPROVED** [review]

**Review by merspieler — APPROVED** [review]
Looks good to me

**Comment by legoboyvdlp** [comment]
Wait!

**Comment by merspieler** [comment]
What? You've approved the changes

**Comment by legoboyvdlp** [comment]
That was just the code itself, I always do a code review in git then a functional review.
Sorry - I never said that I suppose, no worries, I did find some things but not breaking bugs.

**Pull request merged**